app/assets/javascripts/qsets.js

```diff
@@ -94,3 +94,11 @@ function initQuestionFilter() {
     $('.no-questions > p').html(noQuestionsNotification);
   }
 }
+
+function showNoSubsetNotification() {
```

Comment:
Technically, we don't have to do this in JS at all, right? Adding or removing qsets is something that currently requires refreshing the page anyway, so this could all be done (more simply, I think) directly in the view, e.g. with 
```
<% if @qsets.length == 0 %>
```
... I think.

The reason it's done in JS for the questions message is because you can toggle back and forth between the different filter types, and the message changes depending on what the filter is currently. That said, we could've created three different 'no questions' messages on the view/template, one for each filter type, and the client-side JS would just show the appropriate one, but if more filter types were needed in the future that might not scale very well, so that's probably why we went with building the message using JS.

You would also need JS if you were doing add/removes via AJAX, such that the server didn't re-render the page each time. That's not the case here -- I think qsets can't be added or removed from the page dynamically, they require a full page reload.

So if I'm right about that, then the `noSubsetsNotification` message could be generated server-side and be included in the template here https://github.com/penzance/askup/pull/84/files#diff-a5fcfadee9e6acb9ff9a51c88bb9816bR17, and that `<div class="box valign no-qsets hidden row">` section can be wrapped in an `<% if %>` statement. That would be quicker and cleaner, require less JS delivered to the user, and reduce client-side complexity (it would keep the logic closer to the actual generation of the server-side `@qsets` variable).

I may have missed something here, but perhaps this can be much simpler. Let me know what you think @rckim77.